docs/design/factories.md

```diff
@@ -0,0 +1,234 @@
+# Instantiating New Provider-Controlled Types
+
+Historically, the framework has always tried to use providers, data sources,
+and resources that are scoped to the RPC server; you can see this in the SDKv2
+[`ProviderFunc`][sdkv2-provider-func] type, which is used to instantiate a
+provider when the server starts up. But you can also see it in the practice of
+[using a function][sdkv2-resource-func-call] to [register resource
+schemas][sdkv2-resource-schema], even though there's no enforced requirement
+that providers do this. This keeps the variables representing providers, data
+sources, and resources scoped to the gRPC server, which is very helpful when
+running multiple servers at the same time, as in the acceptance test drivers.
+
+There are two main goals when instantiating new instances of a type: to provide
+isolation from other instances of that type that may be running (otherwise, a
+global can just be used) and to (sometimes) allow the provider to populate some
+values on that type (e.g., filling in a schema).
+
+There are a few ways these values can be instantiated, and some of the details
+are dependent on whether the types in question are [framework-controlled or
+provider-controlled][structs-interfaces].
+
+This document is meant to catalogue our available options and explore their
+benefits and trade-offs.
+
+## Options
+
+### Anonymous functions
```

Comment:
It might be clearer if instead of "anonymous functions" and "named functions" (which to me sounds like a distinction between ways of declaring a function - whether we bind an identifier to the function itself), we used the terms "anonymous function types" and "named function types".